src/Autofac/Util/LinkerAttributes.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Autofac Project. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#pragma warning disable SA1649 // File name should match first type name
+#pragma warning disable SA1402 // File may only contain a single type
+#if !NET5_0
```

Comment:
We'll have to be careful about this as we add/remove TFMs to the project. We did something similar with `[Serializable]` [back in the early netstandard days](https://github.com/autofac/Autofac/issues/782) and when there were NuGet bugs such that the build was referencing the wrong TFM folks got "duplicate attribute" errors. It was pretty hard to track down. If a .NET 5 project accidentally ends up referencing (directly or indirectly) the `netstandard2.0` version of Autofac, for example, then someone might see that error.

Reply:
For this one I sort of followed the same pattern as the [NullableAttributes](https://github.com/autofac/Autofac/blob/develop/src/Autofac/Util/NullableAttributes.cs) added for NRT compatibility shims.

I suppose there's an argument that I should use `#if NETSTANDARD2_0 || NETSTANDARD2_1` rather than `#if NET5_0` though.

Reply:
I don't think it'd fix the problem, which is that a transient dependency chain not _quite_ working right would end up referencing the wrong framework's version of the assembly and the shims would show up in a .NET 5 project. But I think they're a lot further down the road with this than they were when we ran into it before, so let's leave it and see what happens. Just something to be aware of in case you see an issue about it.